scaleway/resource_k8s_cluster_beta.go

```diff
@@ -281,6 +339,28 @@ func resourceScalewayK8SClusterBetaCreate(d *schema.ResourceData, m interface{})
 
 	req.AutoscalerConfig = autoscalerReq
 
+	autoUpgradeEnable, okAutoUpgradeEnable := d.GetOk("auto_upgrade.0.enable")
+	autoUpgradeStartHour, okAutoUpgradeStartHour := d.GetOk("auto_upgrade.0.maintenance_window_start_hour")
+	autoUpgradeDay, okAutoUpgradeDay := d.GetOk("auto_upgrade.0.maintenance_window_day")
+
+	// check if either all or none of the auto upgrade attribute are set.
+	// if one auto upgrade attribute is set, they all must be set.
+	// if none is set, auto upgrade attributes will be computed.
+	// We either want all true (all set) or all false (none set), so they must be equal
+	if okAutoUpgradeDay == okAutoUpgradeEnable && okAutoUpgradeEnable == okAutoUpgradeStartHour {
```

Comment:
Do we need to check that `autoUpgradeEnable == true` here?

nit: `if okAutoUpgradeEnable == okAutoUpgradeDay  && okAutoUpgradeEnable == okAutoUpgradeStartHour {`

Reply:
Nop, no need to check, it's done just after